testsuite/classlibrary/TestPattern.sc

```diff
@@ -155,6 +155,21 @@ TestPattern : UnitTest {
 		this.assert(x.sum { |x| x.delta } == 4.5, "Psync with maxdur = quant should end no earlier than after dur");
 	}
 
+	test_Pfset_evaluates_init_and_cleanup_on_empty_stream {
+		var x = 0, y = 0, inEvent = (), cleanup = EventStreamCleanup.new;
+		var outEvent = Pfset({ x = 1 }, p{}, { y = 2 }).asStream.next(inEvent);
```

Comment:
Ok, I agree with this.

> If Pfset has a bug and doesn't change oe (outEvent) back to nil, that would be a bug in Pfset we'd want to detect. But we would not be able to do that if we were to let oe (outEvent) have the initial default value nil by not initializing it to something else than nil before calling Pfset.

As I guess you realized, Pfset is not responsible for setting the variable -- the assignment operator is. If Pfset's stream yields a value (including nil), it will be assigned to outEvent -- and the only way that won't happen is if the stream throws an error (by definition a failure). So no previous value is relevant.

Reply:
Thank you for your patience with all my newbie errors!